terraform/modules/quorum-cluster-region/quorum-cluster-region-backup-event.tf

```diff
@@ -0,0 +1,399 @@
+  }
+  # destroy
+  provisioner "local-exec" {
+    when = "destroy"
+    command = "rm ${var.aws_region}-${var.BackupLambda_output_path}"
+    on_failure = "continue"
+  }
+  // # creat
+  // provisioner "local-exec" {
+  //   command = "zip -j ${var.aws_region}-${var.BackupLambda_output_path} ${var.BackupLambda_source_file}"
+  // }
+  # the creation is in "archive_file" "BackupLambda"
+}
+
+data "archive_file" "BackupLambda" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  type        = "zip"
+  source_file  = "${var.BackupLambda_source_file}"
+  output_path = "${var.aws_region}-${var.BackupLambda_output_path}"
+  depends_on = ["null_resource.BackupLambda"]
+}
+
+resource "aws_kms_key" "a" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  description = "Used for encrypting SSH keys on S3"
+  tags {
+     name = "BackupLambda-${var.network_id}-${var.aws_region}-KMS"
+  }
+}
+
+output "aws_kms_key_test" {
+  value = "${aws_kms_key.a.*.id}"
+}
+
+# Encrypt the contents of the file located at var.private_key_path
+data "aws_kms_ciphertext" "EncryptSSHKey" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  key_id = "${aws_kms_key.a.id}"  
+  plaintext = "${file("${var.private_key_path}")}"
+}
+
+# Save the encrypted contents to the file specified at filename
+resource "local_file" "EncryptedSSHKey" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  content = "${base64decode("${data.aws_kms_ciphertext.EncryptSSHKey.ciphertext_blob}")}"
+  filename = "${var.enc_ssh_path}-${var.aws_region}"
+}
+
+resource "aws_kms_grant" "a" {
+  count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+  name              = "kms-grant-${var.network_id}-${var.aws_region}"
+  key_id            = "${aws_kms_key.a.key_id}"
+  grantee_principal = "${aws_iam_role.iam_for_BackupLambda.arn}"
+  operations        = ["Encrypt", "Decrypt"]
+}
+
+output "aws_s3_bucket" {
+  value = "${element(concat(aws_s3_bucket.quorum_backup.*.id, list("")), 0)}"
+}
+
+resource "aws_security_group" "allow_all" {
+    count = "${signum(lookup(var.maker_node_counts, var.aws_region, 0))}"
+    name        = "BackupLambdaSSH-${var.network_id}-${var.aws_region}-allow_all"
+    description = "Allow all outgoing traffic"
+    vpc_id = "${aws_vpc.quorum_cluster.id}"
+
+    egress {
+        from_port   = 0
+        to_port     = 0
+        protocol    = "-1"
+        cidr_blocks = ["0.0.0.0/0"]
+        description = "Allow all traffic"
+    }
+    ingress {
+      from_port = 22
+      to_port = 22
+      protocol = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+      description = "Allow all SSH traffic"
+    }
+  tags {
+     name = "BackupLambda-${var.network_id}-${var.aws_region}-SG"
+  }
+}
+
+// use the next value after data.template_file.quorum_observer_cidr_block
+data "template_file" "quorum_maker_cidr_block_lambda" {
```

Comment:
If this is just for the lambda we don't need a big network. Maybe we should split this CIDR into a few subnets and just assign this to the first one?